app/gui/view/graph-editor/src/new_node_position.rs

```diff
@@ -0,0 +1,199 @@
+//! This module provides functions returning positions for newly created nodes.
+//!
+//! The returned positions are such that the new nodes will not overlap with existing ones.
+
+use crate::prelude::*;
+
+use crate::component::node;
+use crate::free_place_finder::find_free_place;
+use crate::free_place_finder::OccupiedArea;
```

Comment:
I think the `free_place_finder` module can be made a submodule of `new_node_position`. So far, the finder is used solely for searching for new node positions.